### ISSUE TITLE
Create new entity for AMI overrides

### ISSUE THREAD

**Issue by kathyccheng** [bloom-housing/bloom#1557]

**Comment by kathyccheng** [comment]
@seanmalbert can you add in more details here?

**Comment by seanmalbert** [comment]
Per #1544 , we want a way to override values that get calculated for the unit summary. We'll need an entity that has an amiChartId column to reference the ami chart, a unitId column to reference the units, and then for each household size that is overwritten we would have the max annual income value.

So an ami charts and units would both have a one to many relationship with this new entity.

**Comment by pbn4** [comment]
@seanmalbert 

If I understand correctly you want to be able to to override max annual income for specific household sizes, correct?

Right now ami_chart table contains a jsonb column with AmiChartItems:

```
@Entity()
export class AmiChart {
...
  @Column("jsonb")
  @Expose()
  @IsDefined({ groups: [ValidationsGroupsEnum.default] })
  @ValidateNested({ groups: [ValidationsGroupsEnum.default], each: true })
  @Type(() => AmiChartItem)
  items: AmiChartItem[]
...
}
```

so there is no way to reference a specific AmiChartItem directly by ID, but we could provide an AmiChartItem array the same way it is defined in AmiChart and let the backend merge those two based on `percentOfAmi` and `householdSize`. 

Creating another table here for ami_chart_items is going to incur even more performance problems on /listings endpoint and those ami_chart_items were so far never modified separately from AmiChart. What do you think @seanmalbert ?

**Comment by seanmalbert** [comment]
@pbn4 ,
That's right, I forgot that it needs to also reference the percentOfAmi selected and the chart selected. What if just for ami chart, we update the relationship on listings so that it only stores the chart id (like you were suggesting) and the items that correlate to the percentOfAmi. We would still fetch the charts for the form as we do now (which is a separate call), so we can create the override fields on the frontend once the user enters a percentOfAmi. And we can just store those items, which would be a much smaller set than the whole current one.

**Comment by kathyccheng** [comment]
Should reference overrides in unit summary table
